### PR TITLE
Update go.lang.security.audit.crypto.bad_imports rules

### PR DESCRIPTION
**Pull request by ccrims0n**
* Add pattern-inside the import to avoid confusion with similar named
variables

To solve: https://semgrep.dev/s/2xWL

### CODE REVIEW

**Comment by aryx** [comment]
I've added your on our list of contributors!

**Comment by aryx** [comment]
@cla-bot check

**Comment by aryx** [comment]
Yep it worked! Thx a lot for your contribution!
I'll let one of the maintainer of semgrep-rules/ review, but LGTM. This should remove some false positives.

File: go/lang/security/audit/crypto/bad_imports.yaml [hunk 1]
```diff
@@ -11,15 +11,34 @@ rules:
...
+      - pattern-inside:
+          import "crypto/rc4"
+          ...
+      - pattern: |
+          rc4.$FUNC(...)
+    - patterns:
+      - pattern-inside:
+          import "net/http/cgi"
+          ...
+      - pattern: |-
```
Comment (ccrims0n) [inline go/lang/security/audit/crypto/bad_imports.yaml]: Need to fix this and multi line patterns

**Review by inkz — APPROVED** [review]
lgtm!
@ccrims0n thank you!

**Comment by ccrims0n** [comment]
@inkz Who can help me merge this?

**Comment by aryx** [comment]
@minusworld any idea why CI takes forever on this PR? Is it because it's from a non-r2cerr? But they why it didn't fail earlier?

**Comment by minusworld** [comment]
I'm not certain what's happening here 🤔  I'll take a look soon

**Comment by minusworld** [comment]
Hmm, this is certainly odd... other PRs are working. Let me pull this branch down and poke at it.

**Comment by minusworld** [comment]
That was strange - the rule was missing a pipe operator `|` after `pattern-inside`. Apparently even without the pipe it was valid YAML and caused everything to go very awry. Generally, the YAML parser will detect this and complain. 🤷‍♂️  I'm not sure why it wasn't picked up as an error this time.

In any case, I added in that pipe and things should be good now. 👍

**Comment by aryx** [comment]
Looks like a good additonal check for the metachecker!

**Comment by minusworld** [comment]
> What was the error? Did we get a parsing error or something?

This was the error. It was totally unheplful:

```
➜  crypto git:(update_go_crypto_rules) semgrep --test
A new version of Semgrep is available. Please see https://github.com/returntocorp/semgrep#upgrading for more information.
 25%|████████████████████████                                                                        |1/4Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 576, in _handle_results
    task = get()
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/connection.py", line 251, in recv
    return _ForkingPickler.loads(buf.getbuffer())
```

And then Semgrep would hang indefinitely, I suppose waiting for this thread to finish.

I noticed the missing pipe with my eyeballs. 👀  😆

**Comment by minusworld** [comment]
@ccrims0n Thanks for the update! I'll merge this is now.

**Comment by ccrims0n** [comment]
Thanks @minusworld for the pipe fixed. Got missed -_-

**Comment by aryx** [comment]
cc @emjin for the discussion about metachecker; would be good if the metachecker flag the rule not using the |
(not sure why it worked without though).

**Pull request merged**